src/internal/passwordHash.ts

```diff
@@ -0,0 +1,12 @@
+import * as Scrypt from "@noble/hashes/scrypt"
+import * as Data from "effect/Data"
+import * as Effect from "effect/Effect"
+
+class ScryptOptionsError extends Data.Error<{ message: string }> {}
```

Comment:
Just make it this:

```ts
{
  cause: unknown
}
```

Also, let's use `Data.TaggedError` (https://effect.website/docs/error-management/yieldable-errors#datataggederror)

Reply:
Also, if we keep the `internal/*` modules, then this should just be named `OptionsError` (`Scrypt.OptionsError`).